services/src/machine_learning/xgb_config_builder.rs

```diff
@@ -0,0 +1,1057 @@
+            TreeMethod::GpuHist => "gpu_hist".to_owned(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub enum TreeUpdater {
+    GrowColMaker,
+    DistCol,
+    GrowHistMaker,
+    GrowLocalHistMaker,
+    GrowSkMaker,
+    Sync,
+    Refresh,
+    Prune,
+}
+
+impl ToString for TreeUpdater {
+    fn to_string(&self) -> String {
+        match *self {
+            TreeUpdater::GrowColMaker => "grow_colmaker".to_owned(),
+            TreeUpdater::DistCol => "distcol".to_owned(),
+            TreeUpdater::GrowHistMaker => "grow_histmaker".to_owned(),
+            TreeUpdater::GrowLocalHistMaker => "grow_local_histmaker".to_owned(),
+            TreeUpdater::GrowSkMaker => "grow_skmaker".to_owned(),
+            TreeUpdater::Sync => "sync".to_owned(),
+            TreeUpdater::Refresh => "refresh".to_owned(),
+            TreeUpdater::Prune => "prune".to_owned(),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+pub enum ProcessType {
+    #[default]
+    Default,
+    Update,
+}
+
+impl ToString for ProcessType {
+    fn to_string(&self) -> String {
+        match *self {
+            ProcessType::Default => "default".to_owned(),
+            ProcessType::Update => "update".to_owned(),
+        }
+    }
+}
+
+impl<'a> From<&'a str> for ProcessType {
+    fn from(s: &'a str) -> Self {
+        match s {
+            "default" => ProcessType::Default,
+            "update" => ProcessType::Update,
+            // TODO: better error handling
+            _ => panic!("no known process_type for {s}"),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+pub enum GrowPolicy {
+    #[default]
+    Depthwise,
+    LossGuide,
+}
+
+impl ToString for GrowPolicy {
+    fn to_string(&self) -> String {
+        match *self {
+            GrowPolicy::Depthwise => "depthwise".to_owned(),
+            GrowPolicy::LossGuide => "lossguide".to_owned(),
+        }
+    }
+}
+
+impl<'a> From<&'a str> for GrowPolicy {
+    fn from(s: &'a str) -> Self {
+        match s {
+            "depthwise" => GrowPolicy::Depthwise,
+            "lossguide" => GrowPolicy::LossGuide,
+            // TODO: better error handling
+            _ => panic!("no known grow policy for {s}"),
+        }
+    }
+}
+
+#[derive(Clone, Debug, Default, Serialize, Deserialize, PartialEq)]
+pub enum Predictor {
+    #[default]
+    Cpu,
+    Gpu,
+}
+
+impl ToString for Predictor {
+    fn to_string(&self) -> String {
+        match *self {
+            Predictor::Cpu => "cpu_predictor".to_owned(),
+            Predictor::Gpu => "gpu_predictor".to_owned(),
+        }
+    }
+}
+
+impl<'a> From<&'a str> for Predictor {
+    fn from(s: &'a str) -> Self {
+        match s {
+            "cpu_predictor" => Predictor::Cpu,
+            "gpu_predictor" => Predictor::Gpu,
+            // TODO: better error handling
+            _ => panic!("no known predictor for {s}"),
```

Comment:
Implement TryFrom instead of From and return an error

Reply:
see: https://github.com/geo-engine/geoengine/pull/746#discussion_r1252302470